synthesizer/src/process/execute.rs

```diff
@@ -254,17 +255,28 @@ impl<N: Network> Process<N> {
                     }
                 }
 
-                // Retrieve the output registers.
-                let output_registers =
-                    &finalize.outputs().iter().map(|output| output.register().clone()).collect::<Vec<_>>();
+                // Retrieve the output operands.
+                let output_operands =
+                    &finalize.outputs().iter().map(|output| output.operand().clone()).collect::<Vec<_>>();
 
                 // TODO (howardwu): Save the outputs in ProgramStore.
                 // Load the outputs.
-                let _outputs = output_registers
+                let _outputs = output_operands
                     .iter()
-                    .map(|register| {
-                        // Retrieve the stack value from the register.
-                        registers.load(stack, &Operand::Register(register.clone()))
+                    .map(|operand| {
+                        // Load the outputs.
+                        match operand.clone() {
```

Comment:
it would be better to not clone here, and instead apply it in the branches that need it